bridges/ZDNetBridge.php

```
@@ -160,143 +160,44 @@ class ZDNetBridge extends BridgeAbstract {
-				$article = getSimpleHTMLDOM($article_url)
-					or returnServerError('Could not request ZDNet: ' . $article_url);
-
-				if(!empty($article_author)) {
-					$author = $article_author;
-				} else {
-					$author = $article->find('meta[name=author]', 0);
-					if(is_object($author)) {
-						$author = $author->content;
-					} else {
-						$author = 'ZDNet';
-					}
-				}
+		$this->collectExpandableDatas($url);
+	}
 
-				$thumbnail = $article->find('meta[itemprop=image]', 0);
-				if(is_object($thumbnail)) {
-					$thumbnail = $thumbnail->content;
-				} else {
-					$thumbnail = '';
-				}
+	protected function parseItem($item){
+		$item = parent::parseItem($item);
 
-				$contents = $article->find('article', 0)->innertext;
-				foreach(array(
-					'<div class="shareBar"',
-					'<div class="shortcodeGalleryWrapper"',
-					'<div class="relatedContent',
-					'<div class="downloadNow',
-					'<div data-shortcode',
-					'<div id="sharethrough',
-					'<div id="inpage-video'
-				) as $div_start) {
-					$contents = stripRecursiveHtmlSection($contents, 'div', $div_start);
-				}
-				$contents = stripWithDelimiters($contents, '<script', '</script>');
-				$contents = stripWithDelimiters($contents, '<meta itemprop="image"', '>');
-				$contents = trim(stripWithDelimiters($contents, '<section class="sharethrough-top', '</section>'));
-				$content_img = strpos($contents, '<img'); //Look for first image
-				if (($content_img !== false && $content_img < 512) || $thumbnail == '') {
-					$content_img = ''; //Image already present on article beginning or no thumbnail
-				} else {
-					$content_img = '<p><img src="'.$thumbnail.'" /></p>'; //Include thumbnail
-				}
-				$contents = $content_img
-				. '<p><b>'
-				. $article_subtitle
-				. '</b></p>'
-				. $contents;
+		$article = getSimpleHTMLDOMCached($item['uri']);
+		if(!$article) {
+			$item['content'] .= '<p><em>Could not request ZDNet: ' . $url . '</em></p>';
```

Here, you should return an error except if there is a valid reason why one specific element is not fetched. In this case, just don't return the article

---

That was to keep consistency with the failsafe behavior explained [here](https://github.com/RSS-Bridge/rss-bridge/pull/802#discussion_r216135762), in order to keep the bridge working in reduced functionality mode when only item contents fails to be fetched. Is it bad?

---

I am not really sure about this, even with the previous behaviour...
While it makes sense to not fail when we try to fetch an element, I think returnServerError makes more sense syntactically...
I still stand by my original opinion, however  I understand your point. What do you think @LogMANOriginal ?

---

As this is a broader discussion about how to implement bridges (not just this one), I can stick back to `returnServerError` for now and maybe we can discuss this at a later time using a separate issue?

---

Sure.
I'll open it, I wouldn't want to hold this PR any longer.

---

Changed to `returnServerError`.